napari/utils/_pydantic/custom_types.py

```diff
@@ -0,0 +1,34 @@
+import numpy as np
+
+
+class _ArrayMeta(type):
+    def __getitem__(self, t):
+        return type('Array', (Array,), {'__dtype__': t})
+
+
+class Array(np.ndarray, metaclass=_ArrayMeta):
```

Comment:
```suggestion
class Array(np.ndarray):
    def __class_getitem__(cls, t):
        return type('Array', (Array,), {'__dtype__': t})
```

This metaclass strategy would also work with <py3.7, but since 3.7 we have [`__class_getitem__`](https://www.python.org/dev/peps/pep-0560/#specification) which looks better, and lets us avoid using metaclasses and the conflicts that they sometimes cause.

Reply:
wow, that's great!